dataflow_lite/include/dataflow_lite/cloudwatch/cloudwatch_service.h

```diff
@@ -69,6 +70,8 @@ CloudWatchService(std::shared_ptr<Publisher<std::list<T>>> publisher,
   this->file_upload_streamer_ = nullptr;
   this->dequeue_duration_ = kDequeueDuration;
   this->number_dequeued_.store(0);
+  this->last_publish_time_ = getCurrentTimestamp();
```

Comment:
Can there be an edge case where this causes the very first `work()` call to be throttled if it happens under `minimum_publish_period`?

Reply:
Initial value of 0?

Reply:
Yeah - it's possible if everything initializes and batches quickly. In effect this just means that the first actual sending to cloudwatch won't happen until at least `minimum_publish_period_` after initialization, I can't see a case where this effect would realistically change results in any noticeable way.

The way I can think of to avoid it would be to add another branch to check for "first time", perhaps by using a sentinel value to check for. This adds an extra check in every publish, but maybe that's worth it.

Reply:
Yeah, @dabonnie, I'll set it to 0 and check for that.